### PR TITLE
MAINT: stats: add hypotests to __all__ in init.py, not stats.py

### PR DESCRIPTION
**Pull request by mdhaber**
#### Reference issue

gh-13787

#### What does this implement/fix?

We started adding hypothesis tests to `__all__` in `stats.py`. They should be added in `__init__.py`.

#### Additional information

Moved `rvs_ratio_uniforms`, too.

I think we should move `alexandergovern` from the 8700-line `stats.py` to `_hypotests.py` as it's a relatively new hypothesis test and should have been added there initially.

### CODE REVIEW

File: scipy/stats/stats.py [hunk 1]
```diff
@@ -101,12 +101,10 @@
 
    pearsonr
    fisher_exact
-   barnard_exact
```
Comment (mdhaber) [inline scipy/stats/stats.py:104]: This docstring is irrelevant, no? Should we just delete it all?

File: scipy/stats/stats.py [hunk 2]
```diff
@@ -189,13 +182,10 @@
...
 from ._stats import (_kendall_dis, _toint64, _weightedrankedtau,
                      _local_correlations)
-from ._rvs_sampling import rvs_ratio_uniforms
-from ._page_trend_test import page_trend_test
 from dataclasses import make_dataclass
-from ._hypotests import (epps_singleton_2samp, somersd, cramervonmises,
-                         cramervonmises_2samp, barnard_exact)
 
 
+# Functions classes in other files should be added in `__init__.py`, not here
```
Comment (mdhaber) [inline scipy/stats/stats.py:188]:
```suggestion
# Functions/classes in other files should be added in `__init__.py`, not here
```

File: scipy/stats/stats.py (hunk 1 above)
Comment (rgommers) [reply scipy/stats/stats.py:104]: Indeed, it's irrelevant, +1 for deleting.

**Review by rgommers — APPROVED** [review]
LGTM, in it goes. Thanks Matt.

**Pull request merged**